### PR TITLE
chore(filter): Add a switch statement

### LINKED ISSUE #1096: Change filter in query implementation to use a switch
As mentioned in the following pull request:

https://github.com/googleapis/nodejs-datastore/pull/1085

We should `switch` on `arguments.length` in the `filter` function of `query`.

### PR DESCRIPTION
**Pull request by danieljbruce**
In the filter function a switch statement is added so that it is clear to the reader what happens when the arguments are of various lengths.

Fixes #1096

### CODE REVIEW

File: src/query.ts [hunk 1]
```diff
@@ -222,24 +222,39 @@ class Query {
...
-      let operator = operatorOrValue as Operator;
-      if (arguments.length === 2) {
-        value = operatorOrValue as AllowedFilterValueType<T>;
-        operator = '=';
+    }
+    switch (arguments.length) {
+      case 1: {
+        if (isFilter(propertyOrFilter)) {
+          this.entityFilters.push(propertyOrFilter);
+        } else {
```
Comment (vicb) [inline src/query.ts:234]: I think the `else` branch should be removed here (`operatorOrValue` and `value` are not provided)

File: src/query.ts [hunk 2]
```diff
@@ -222,24 +222,39 @@ class Query {
...
+            op: (operatorOrValue as Operator).trim() as Operator,
+            val: value,
+          });
+        }
+        break;
+      }
+      case 2: {
+        this.filters.push({
+          name: (propertyOrFilter as String).trim(),
+          op: '=' as Operator,
```
Comment (vicb) [inline src/query.ts:246]: `as Operator` is probably not needed here?

**Review by vicb — APPROVED** [review]
LGTM with a couple nits

File: src/query.ts (hunk 1 above)
Comment (danieljbruce) [reply src/query.ts]: ok

File: src/query.ts (hunk 2 above)
Comment (danieljbruce) [reply src/query.ts]: yup

**Review by vicb — APPROVED** [review]
Nice!

**Review by ruyadorno — APPROVED** [review]

**Pull request merged**